### PR TITLE
Initial docs push

### LINKED ISSUE #47: Create full repo docs and sub-package technical docs (apis)
This is needed as part of our docs push, but is also just a must have in general.

### PR DESCRIPTION
**Pull request by carsonfarmer**
Signed-off-by: Carson Farmer <carson.farmer@gmail.com>

## Description

First attempt to pull in docs and start to build out examples etc for js-threads.

Fixes #47.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change ~requires~_is_ a documentation update

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] ~I have added tests that prove my fix is effective or that my feature works~
- [ ] ~New and existing unit tests pass locally with my changes~
- [ ] ~Any dependent changes have been merged and published in downstream modules~

### CODE REVIEW

**Comment by carsonfarmer** [comment]
I think this is sufficiently comprehensive for the first push of docs now @andrewxhill. There are perhaps too many details in here? But there are sufficiently low down the readme that I think only those who are interested are actually going to bother reading. Additionally, I have simplified all the sub-package readmes, as these will likely only ever be seen on npmjs.org, whereas the top-level one is visible on GitHub etc.

File: packages/core/README.md [hunk 1]
```diff
@@ -1,57 +1,9 @@
...
-[![Chat on Slack](https://img.shields.io/badge/slack-slack.textile.io-informational.svg?style=popout-square)](https://slack.textile.io)
-[![npm (scoped)](https://img.shields.io/npm/v/@textile/threads-core.svg?style=popout-square)](https://www.npmjs.com/package/@textile/threads-core)
-[![docs](https://img.shields.io/badge/docs-master-success.svg?style=popout-square)](https://textileio.github.io/js-threads)
-[![standard-readme compliant](https://img.shields.io/badge/standard--readme-OK-green.svg?style=flat-square)](https://github.com/RichardLitt/standard-readme)
+[![Textile](https://img.shields.io/badge/made%20by-Textile-informational.svg)](https://textile.io)
+[![License](https://img.shields.io/github/license/textileio/js-threads.svg)](../../LICENSE)
+[![npm (scoped)](https://img.shields.io/npm/v/@textile/threads-core.svg)](https://www.npmjs.com/package/@textile/threads-core)
 
-> Core interfaces and classes for Textile's Threads Protocol
+> Core interfaces and classes for Textile's Threads protocol
```
Comment (andrewxhill) [inline packages/core/README.md:7]: it occurred to me that the DB might not be part of the protocol technical. "Threads protocol and database"?

Comment (carsonfarmer) [reply packages/core/README.md:7]: Yes, good call, this should be updated in all 5(?) places, good point.

**Comment by carsonfarmer** [comment]
Cool, actually snuck in that fix :)

**Pull request merged**